contrib/scripts/check_oid_ovfps.sh

```diff
@@ -0,0 +1,515 @@
+declare IFS_backup=$IFS
+
+current_dir=""
+work_dir=""
+
+######################################################################
+# functions
+#
+function echo_stderr()
+{
+#   echo "$@" 1>&2;
+#   cat <<< "$@" 1>2&;
+    printf "%s\n" "$*" 1>&2;
+}
+
+function show_usage ()
+{
+	echo "Usage: $0 [OPTIONS] [database]"
+	echo " OPTIONS"
+	echo "  -C client-server mode execution, default"
+	echo "  -S standalone mode execution"
+	echo "  -u, --user=ARG      Set database user name(DBA or DBA group member); default dba"
+	echo "  -p, --password=ARG  Set password of databases user name; default NULL"
+	echo "  -o, --owner=ARG     Set target owner ID; default: NULL"
+	echo "  -c, --class=ARG     Set target class name; default: NULL"
+	echo "  -k, --key=ARG       Set the name of the key field to be checked; default: Max_num_ovf_page_a_key"
+        echo "                      Choose from Max_num_ovf_page_a_key, Avg_num_ovf_page_per_key, Num_ovf_page or Avg_num_value_per_deduplicate_key."
+	echo "  -t, --threshold=ARG Set the threshold to select the index to be reported; default: 1000"    
+	echo "  -s, --sort=ARG      Set sort filed name(pages, owner, table, index); default pages"
+	echo ""
+	echo " EXAMPLES"
+	echo "  $0 -S -o user1 demodb"
+	echo "  $0 -C -c game -t 100 demodb"
+	echo "  $0 -C -o user1 --key=Max_num_ovf_page_a_key --sort=table demodb"
+	echo ""
+	exit 1
+}
+
+function get_options()
+{
+	local sa_mode=0
+	local tkey=""
+
+	args=$(getopt -o :u:p:o:c:t:k:s:CS  -l user:,password:,owner:,class:,threshold:,key:,sort:   -- "$@" )
+	eval set -- "$args"
+	while true
+	do
+		case "$1" in
+			-C ) (( sa_mode++ ));  connection_mode="-C"  ;;
+			-S ) (( sa_mode++ ));  connection_mode="-S" ;;
+			-u | --user)  user="-u $2";       shift  ;;
+			-p | --password ) pass="-p $2";   shift ;;
+			-o | --owner) target_owner="$2";  shift ;;
+			-c | --class) target_table="$2";  shift ;;
+			-t | --threshold) threshold_value=$2; shift ;;
+			-k | --key)
+				tkey=${2,,}  # to lowercase
+				for (( i =0; i < ${#key_title[@]}; i+=2 ))
+				do
+					if [ x"$tkey" == x"${key_title[${i}]}" ]; then
+						match_key="${sort_title[${i} + 1]}"
+						break
+					fi
+				done
+				if [ $i -gt ${#key_title[@]} ]; then
+					show_usage
+				fi
+				shift ;;
+				
+			-s | --sort)
+				tkey=${2,,} # to lowercase
+				for (( i =0; i < ${#sort_title[@]}; i+=3 ))
+				do
+					if [ x"$tkey" == x"${sort_title[${i}]}" ] || [ x"$tkey" == x"${sort_title[${i} + 1]}" ]; then
+						sort_field=${sort_title[${i} + 2]}
+						break
+					fi
+				done
+				if [ $i -gt ${#sort_title[@]} ]; then
+					show_usage
+				fi
+				shift ;;
+				
+			-- ) shift;  break ;;
+			* )  echo "unknown option"; show_usage ;;
+		esac
+		shift
+	done
+
+	if [ $# -ne 1 ]; then
+		show_usage
+	elif [ $sa_mode -gt 1 ]; then
+		show_usage
+	fi
+
+	database="$@"
+}
+
+function print_args()
+{
+#    echo arguments to stderr
+	echo_stderr "Args list..."
```

Comment:
Progress is printing to stderr, but it would be nice to print it to stdout by supporting -v option.

Reply:
OK, I will fix it.